.ci/Makefile

```
@@ -0,0 +1,164 @@
+TAGS ?= fleet_mode_agent
+
+# Check that given variables are set and all have non-empty values,
+# die with an error otherwise.
+#
+# Params:
+#   1. Variable name(s) to test.
+#   2. (optional) Error message to print.
+check_defined = \
+    $(strip $(foreach 1,$1, \
+        $(call __check_defined,$1,$(strip $(value 2)))))
+__check_defined = \
+    $(if $(value $1),, \
+      $(error Undefined $1$(if $2, ($2))))
+
+@:$(call check_defined, AWS_ACCESS_KEY_ID, You need to export AWS_ACCESS_KEY_ID to create AWS resources under that account)
+@:$(call check_defined, AWS_SECRET_ACCESS_KEY, You need to export AWS_SECRET_ACCESS_KEY to create AWS resources under that account)
+
+# Create the run id if it does not exist
+.runID:
+	echo "$(shell uuidgen|cut -d'-' -f1)" > $(RUN_ID_FILE)
+
+.PHONY: setup-env
+setup-env:
+	$(MAKE) .runID
+	python3 -mvenv $(VENV_DIR)
+	$(VENV_DIR)/bin/pip3 install ansible requests boto3 boto
+	$(VENV_DIR)/bin/ansible-galaxy install -r $(PROJECT_DIR)/.ci/ansible/requirements.yml
+
+.PHONY: destroy-env
+destroy-env:
+	rm -fr $(VENV_DIR)
+
+.PHONY: provision-stack
+provision-stack: .runID
+	@:$(call check_defined, RUN_ID, You need to an unique RUN_ID. To create it please run 'make .runID' goal)
+	$(VENV_DIR)/bin/ansible-playbook $(PROJECT_DIR)/.ci/ansible/playbook.yml \
+		--private-key="$(SSH_KEY)" \
+		--extra-vars "nodeLabel=stack nodeImage=$(STACK_IMAGE) nodeInstanceType=$(STACK_INSTANCE_TYPE) nodeUser=$(STACK_USER)" \
+		--extra-vars "runId=$(RUN_ID) workspace=$(PROJECT_DIR)/ sshPublicKey=$(SSH_KEY_PUBLIC)" \
+		--ssh-common-args='-o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null' \
+		-t provision-stack
+
+.PHONY: setup-stack
+setup-stack: .runID
+	@:$(call check_defined, RUN_ID, You need to an unique RUN_ID. To create it please run 'make .runID' goal)
+	@:$(call check_defined, STACK_IP_ADDRESS, IP address of the stack)
+	$(VENV_DIR)/bin/ansible-playbook $(PROJECT_DIR)/.ci/ansible/playbook.yml \
+		--private-key="$(SSH_KEY)" \
+		--extra-vars "nodeLabel=stack nodeImage=$(STACK_IMAGE) nodeInstanceType=$(STACK_INSTANCE_TYPE) nodeUser=$(STACK_USER)" \
+		--extra-vars "runId=$(RUN_ID) workspace=$(PROJECT_DIR)/ sshPublicKey=$(SSH_KEY_PUBLIC)" \
+		--ssh-common-args='-o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null' \
+		-t setup-stack \
+		-i $(STACK_IP_ADDRESS),
+
+.PHONY: create-stack
+create: provision-stack setup-stack
+
+.PHONY: destroy-stack
+destroy-stack:
+	$(VENV_DIR)/bin/ansible-playbook $(PROJECT_DIR)/.ci/ansible/playbook.yml \
+		--private-key="$(SSH_KEY)" \
+		--extra-vars="nodeLabel=stack nodeImage=$(STACK_IMAGE) nodeUser=$(STACK_IMAGE)" \
+		--extra-vars="runId=$(RUN_ID) workspace=$(PROJECT_DIR)/ sshPublicKey=$(SSH_KEY_PUBLIC)" \
+		--ssh-common-args='-o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null' \
+		-t destroy
+	rm -fr $(PROJECT_DIR)/.ci/stack-host-ip
+
+.PHONY: ssh-stack
+ssh-stack:
+	ssh -i $(SSH_KEY) $(STACK_USER)@$(STACK_IP_ADDRESS) -vvv
```

[Suggestion]
I don't see much use for `-vvv` rather than to debug a connection problem. What is quite rare, so I'd suggest to remove it.  
```suggestion
	ssh -i $(SSH_KEY) $(STACK_USER)@$(STACK_IP_ADDRESS)
```